force-app/main/default/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -194,11 +196,13 @@ export default class GeFormRenderer extends LightningElement{
 
     _isFormCollapsed = false;
     _shouldInformParent = true;
+    _isInvalidDonorSelected = false;
 
     set selectedDonationOrPaymentRecord(record) {
         if (record.new === true) {
             this.setCreateNewOpportunityInFormState();
         } else if (this.isAPaymentId(record.fields.Id)) {
+            this.validateDonorSelection(record.fields);
```

Comment:
since the call to validate the donor is only in the isAPaymentId block, does that mean the validation would only occur when updating a payment from the modal? What if the user updates the opportunity?

Reply:
Using the update opportunity button essentially copies donation data onto a newly created opportunity, that's existing behaviour I don't think we should change, BDI doesn't try to match because we pre-match in Gift Entry by setting the `recordId` for the opportunity upfront.